modules/c++/nitf/source/ByteProvider.cpp

```diff
@@ -566,11 +562,10 @@ void ByteProvider::getBytes(const void* imageData,
                     imageDataEndRow,
                     startGlobalRowToWrite,
                     imageData,
-                    numPadRowsSoFar,
                     fileOffset,
                     buffers);
 
-            addDES(seg, startRow, buffers);
+            addDES(seg, imageDataEndRow, buffers);
```

Comment:
`addDES()` checks whether `seg` is the last segment and if the passed row index is the last row of that segment. It will never add the DES bytes if `startRow` is passed.